cluster.py

```diff
@@ -46,7 +47,8 @@ class ClusterConfig:
 
 
 class Cluster(RetryingStateMachine, WithContainerConfigs):
-    def __init__(self, cluster_config: ClusterConfig, swarm_agent_config: SwarmAgentConfig):
+    def __init__(self, cluster_config: ClusterConfig, swarm_agent_config: SwarmAgentConfig, swarm_client: SwarmApi):
+        self.swarm_client = swarm_client
```

Comment:
Pass this as part of `cluster_config: ClusterConfig`

Reply:
It is not a configuration

Reply:
Actually if you do the other comment, the cluster won't even have to be aware of this object - the swarm creates the SwarmAgentConfig and it's passed as an opaque object to all agents within the cluster